scripts/install_dev_python_modules.py

```diff
@@ -141,6 +141,8 @@ def main(
     if quiet is not None:
         cmd.append(f'-{"q" * quiet}')
 
+    cmd.append("--config-settings editable_mode=compat")
```

Comment:
how confident are you this is necessary ? This setting is going to get dropped at some point (allegedly) and I thought I took care of all the problems that required setting this. What type of caching issue were you running in to?

Reply:
Not super confident.   But it was pretty frustrating setting up a new venv a couple weeks ago and seeing it use cached wheels.  It took a couple times of blowing away the venv and running it with these options appended that got things working for me.

Reply:
>  seeing it use cached wheels

Can you clarify what you mean here? I will feel better landing this if we know specifically what behavior was problematic that this flag addresses so we can leave a better path for future removal

Reply:
hmm, I just tried again with a fresh venv and it looks like it's behaving as expected without this change.

I'll abandon.

I was seeing pretty broken behavior earlier where running `make dev_install` in the `dagster` repo would work as expected, but things would change when running `make dev_install` in the `internal` repo in the same venv, and it wouldn't resolve the packages (would overwrite the local editable installs with the downloaded package).